system/lib/wasmfs/wasmfs.cpp

```diff
@@ -101,6 +102,46 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  std::shared_ptr<File> file = [&] {
+    auto fileTable = FileTable::get();
+    return fileTable[fd]->get().getFile();
+  }();
+
+  struct stat* buffer = (struct stat*)buf;
+
+  auto fileInfo = file->get();
+
+  if (file->is<Directory>()) {
+    buffer->st_size = 4096;
+  } else if (file->is<DataFile>()) {
+    buffer->st_size = fileInfo.size();
+  } else { // TODO: add size of symlinks
+    buffer->st_size = 0;
+  }
+
+  // ATTN: hard-coded constant values below match native system compatibility.
```

Comment:
Are these copied from the JS code, or somewhere else?

Reply:
Yep, I can make this clearer in the comment, also the other comment as well.